src/GafferCortex/CompoundParameterHandler.cpp

```diff
@@ -84,24 +84,24 @@ void CompoundParameterHandler::restore( Gaffer::GraphComponent *plugParent )
 		ParameterHandler *h = handler( it->get(), true );
 		if( h )
 		{
-			h->restore( compoundPlug.get() );
+			h->restore( compoundPlug );
 		}
 	}
 
 }
 
 Gaffer::Plug *CompoundParameterHandler::setupPlug( Gaffer::GraphComponent *plugParent, Gaffer::Plug::Direction direction, unsigned flags )
 {
-	// decide what name our compound plug should have
+	// decide what name our plug should have
 
 	std::string name = plugName();
 
-	// create the compound plug if necessary
+	// create the plug if necessary
 
-	m_plug = plugParent->getChild<Gaffer::CompoundPlug>( name );
+	m_plug = plugParent->getChild<Gaffer::Plug>( name );
```

Comment:
I think that this change would perhaps make https://github.com/ImageEngine/jabuka/pull/2058 unnecessary, since the handler would now have a generic Plug, which would then not require, in the handler's `setPlugValue()`, that all the children of the plug are themselves ValuePlugs.
The Jabuka's PR should still be necessary for now, so that it works with current gaffer versions.
And, having Jabuka's connection plugs be ValuePlugs is not unreasonable.

Reply:
That's good to know. As I was saying on the other ticket, supporting CompoundPlug meant relaxing some constraints in ValuePlug which has made it easy for problems to exist. Reintroducing those constraints forced me into the changes you see here, so it's good to know that that ripples through and fixes problems in Jabuka. The end result should be that any logic errors are detected much sooner (typically when trying to make an invalid parent/child relationship)...